databricks/koalas/tests/test_series.py

```diff
@@ -1980,3 +1981,53 @@ def test_first_valid_index(self):
         pser = pd.Series([])
         kser = ks.from_pandas(pser)
         self.assert_eq(pser.first_valid_index(), kser.first_valid_index())
+
+    @unittest.skipIf(
+        LooseVersion(pyspark.__version__) < LooseVersion("2.4"),
+        "explode won't work properly with PySpark<2.4",
+    )
```

Comment:
Oh, the tests was failed with `pyspark.sql.utils.AnalysisException` as below so I thought that was issue with PySpark version.

```python
E               pyspark.sql.utils.AnalysisException: "Generators are not supported when it's nested in expressions, but got: generatorouter(explode(0)) AS `__none__`;"
```

But it seems work properly explode itself in PySpark < 2.4. 

Let me give some time to investigate. Thanks!!

Reply:
I think maybe because of https://issues.apache.org/jira/browse/SPARK-24488 is not resolved in PySpark 2.3 ?

Reply:
Then we should just avoid the issue:

```diff
-        if LooseVersion(pyspark.__version__) < LooseVersion("2.4"):
-            raise RuntimeError("explode can be used in PySpark >= 2.4")
         if not isinstance(self.spark.data_type, ArrayType):
             return self.copy()

-        scol_name = self._internal.data_spark_column_names[0]
-        scol = F.explode_outer(self.spark.column).alias(scol_name)
+        scol = F.explode_outer(self.spark.column).alias(name_like_string(self._column_label))

         internal = self._kdf._internal.copy(
-            column_labels=[self._column_label],
-            data_spark_columns=[scol.alias(name_like_string(self._column_label))],
-            column_label_names=None,
+            column_labels=[self._column_label], data_spark_columns=[scol], column_label_names=None
         ).resolved_copy

         return first_series(DataFrame(internal))
```

Reply:
Btw, we should also reset `NATURAL_ORDER_COLUMN_NAME`; otherwise the value will be duplicated:

```py
>>> kser = ks.Series([[1, 2, 3], [], [3, 4]])
>>> kser.explode()._internal.spark_frame.show()
+-----------------+--------+-----------------+
|__index_level_0__|__none__|__natural_order__|
+-----------------+--------+-----------------+
|                0|       1|      25769803776|
|                0|       2|      25769803776|
|                0|       3|      25769803776|
|                1|    null|      60129542144|
|                2|       3|      94489280512|
|                2|       4|      94489280512|
+-----------------+--------+-----------------+
```

Reply:
Oh, Got it. Thanks!! :)